src/events.rs

```diff
@@ -159,6 +162,101 @@ pub struct Event {
     pub sig: String,
 }
 
+#[derive(Error, Debug, Eq, PartialEq)]
+pub enum EventError {
+    #[error("Secp256k1 Error: {}", _0)]
+    Secp256k1Error(secp256k1::Error),
+}
+
+impl From<secp256k1::Error> for EventError {
+    fn from(err: secp256k1::Error) -> Self {
+        Self::Secp256k1Error(err)
+    }
+}
+
+impl Event {
+    /// get_content returns the content of the event
+    /// # Example
+    /// ```rust
+    /// use nostr_rust::{events::EventPrepare, utils::get_timestamp, Identity};
+    /// use std::str::FromStr;
+    ///
+    /// let actual_time = get_timestamp();
+    /// let identity = Identity::from_str(env!("SECRET_KEY")).unwrap();
+    /// let event = EventPrepare {
+    ///    pub_key: env!("PUBLIC_KEY").to_string(),
+    ///    created_at: get_timestamp(),
+    ///    kind: 0,
+    ///    tags: vec![],
+    ///    content: "content".to_string(),
+    /// }.to_event(&identity, 0);
+    ///
+    /// assert_eq!(event.get_content(), format!("[0,\"c5aec31e83bdf980939b5ef7c6bcaa2be8bd39d38667da58ba6dba240eb8b69d\",{},0,[],\"content\"]", actual_time));
```

Comment:
Because the tests will not have the same private key as yours, the format function of your assert should also add the public key dynamically